sdk/eventhub/Azure.Messaging.EventHubs/tests/EventHubConsumer/EventHubConsumerLiveTests.cs

```diff
@@ -218,5 +218,327 @@ await using (var consumer = client.CreateConsumer(partition, EventPosition.Lates
+                        }
+
+                        index = 0;
+
+                        foreach (var receivedEvent in receivedEvents)
+                        {
+                            Assert.That(receivedEvent.IsEquivalentTo(eventBatch[index]), Is.True, $"The received event at index: { index } did not match the sent batch.");
+                            ++index;
+                        }
+
+                        Assert.That(index, Is.EqualTo(eventBatch.Length), "The number of received events did not match the batch size.");
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        ///   Verifies that the <see cref="EventHubConsumer" /> is able to
+        ///   connect to the Event Hubs service and perform operations.
+        /// </summary>
+        ///
+        [Test]
+        [TestCase(true)]
+        [TestCase(false)]
+        public async Task ConsumerCannotReceiveWhenClosed(bool sync)
+        {
+            await using (var scope = await EventHubScope.CreateAsync(1))
+            {
+                var connectionString = TestEnvironment.BuildConnectionStringForEventHub(scope.EventHubName);
+
+                var eventBatch = new[] { new EventData(Encoding.UTF8.GetBytes("This message won't be sent")) };
+
+                await using (var client = new EventHubClient(connectionString))
+                {
+                    var partition = (await client.GetPartitionIdsAsync()).First();
+
+                    await using (var consumer = client.CreateConsumer(partition, EventPosition.Latest))
+                    {
+                        Assert.That(async () => await consumer.ReceiveAsync(1, TimeSpan.Zero), Throws.Nothing);
+
+                        if (sync)
+                        {
+                            consumer.Close();
+                        }
+                        else
+                        {
+                            await consumer.CloseAsync();
+                        }
+
+                        Assert.ThrowsAsync<ObjectDisposedException>(async () => await consumer.ReceiveAsync(1, TimeSpan.Zero));
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        ///   Verifies that the <see cref="EventHubConsumer" /> is able to
+        ///   connect to the Event Hubs service and perform operations.
+        /// </summary>
+        ///
+        [Test]
+        public async Task ConsumerCannotReceiveFromInvalidPartition()
+        {
+            await using (var scope = await EventHubScope.CreateAsync(1))
+            {
+                var connectionString = TestEnvironment.BuildConnectionStringForEventHub(scope.EventHubName);
+
+                await using (var client = new EventHubClient(connectionString))
+                {
+                    // Some invalid partition values. These will fail on the service side.
+                    var invalidPartitions = new[]
+                    {
```

Comment:
Can we please move these to `[TestCase]` arguments?   Embedding data in this way makes understanding a failure more difficult, and goes against the "test only one thing" guidance.

Reply:
Done.